File: sdks/insuretech-go-sdk/pkg/models/individual_beneficiary.go
package models

import (
	"time"
)

// IndividualBeneficiary represents a individual_beneficiary
type IndividualBeneficiary struct {
	AuditInfo *AuditInfo `json:"audit_info,omitempty"`
	BeneficiaryId string `json:"beneficiary_id,omitempty"`
	BirthCertificateNumber string `json:"birth_certificate_number,omitempty"`
	ContactInfo *ContactInfo `json:"contact_info,omitempty"`
	DateOfBirth time.Time `json:"date_of_birth,omitempty"`
	FullName string `json:"full_name,omitempty"`
	FullNameBn string `json:"full_name_bn,omitempty"`
	Gender *BeneficiaryGender `json:"gender,omitempty"`
	MaritalStatus *MaritalStatus `json:"marital_status,omitempty"`
	NidNumber string `json:"nid_number,omitempty"`
	NomineeName string `json:"nominee_name,omitempty"`
	NomineeRelationship string `json:"nominee_relationship,omitempty"`
	Occupation string `json:"occupation,omitempty"`
	PassportNumber string `json:"passport_number,omitempty"`
	PermanentAddress *Address `json:"permanent_address,omitempty"`
	PresentAddress *Address `json:"present_address,omitempty"`
	TinNumber string `json:"tin_number,omitempty"`
}

File: backend/insurance_engine/src/InsuranceEngine.Beneficiary/Domain/Entities/IndividualBeneficiary.cs
using InsuranceEngine.Beneficiary.Domain.Enums;
using InsuranceEngine.SharedKernel.Domain;
using InsuranceEngine.SharedKernel.Domain.ValueObjects;

namespace InsuranceEngine.Beneficiary.Domain.Entities;

public class IndividualBeneficiary : Entity<Guid>
{
    public Guid BeneficiaryId { get; private set; }
    
    public string FullName { get; set; } = string.Empty;
    public string? FullNameBn { get; set; }
    public DateTime DateOfBirth { get; set; }
    public BeneficiaryGender Gender { get; set; }
    
    public string? NidNumber { get; set; }
    public string? PassportNumber { get; set; }
    public string? BirthCertificateNumber { get; set; }
    public string? TinNumber { get; set; }
    
    public MaritalStatus MaritalStatus { get; set; }
    public string? Occupation { get; set; }
    
    public string? NomineeName { get; set; }
    public string? NomineeRelationship { get; set; }
 
    public ContactInfo ContactInfo { get; set; } = new();
    public Address PermanentAddress { get; set; } = new();
    public Address PresentAddress { get; set; } = new();
    public AuditInfo AuditInfo { get; private set; } = new();

    // EF Core constructor
    public IndividualBeneficiary() { }

    public IndividualBeneficiary(
        Guid id,
        Guid beneficiaryId,
        string fullName,
        DateTime dob,
        BeneficiaryGender gender) : base(id)
    {
        BeneficiaryId = beneficiaryId;
        FullName = fullName;
        DateOfBirth = dob;
        Gender = gender;
        AuditInfo = new AuditInfo();
    }

    public static IndividualBeneficiary Create(Guid id, Guid beneficiaryId)
    {
        return new IndividualBeneficiary
        {
            Id = id,
            BeneficiaryId = beneficiaryId,
            AuditInfo = new AuditInfo()
        };
    }
}
